### PR TITLE
Windows/v3

### PR DESCRIPTION
**Pull request by victorjulien**
Fixes and updates to support running on Windows. Tested with mingw32 and mingw64 on python2 and python3.

Also adds test for Suricata bug 2769.

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -621,7 +638,10 @@ def default_args(self):
         # Find rules.
         rules = glob.glob(os.path.join(self.directory, "*.rules"))
         if not rules:
-            args += ["-S", "/dev/null"]
+            if sys.platform == "win32":
+                args += ["--disable-detection"]
+            else:
+                args += ["-S", "/dev/null"]
```
Comment (jasonish) [inline run.py]: Any reason not to just use --disable-detection for both cases here?

Comment (victorjulien) [reply run.py]: We could do that. The code paths are different though, so I guess it makes some sense to test both. In the past we've had large perf differences between the opts.

Comment (victorjulien) [reply run.py]: I guess for the purpose of suricata-verify we can make this change indeed.

Comment (victorjulien) [reply run.py]: See #27

**Comment by victorjulien** [comment]
Replaced by #27

**Pull request closed without merging**